assets/prototype/domain/eventEditor/containers/mutations/useDeleteDateMutation.js

```diff
@@ -1,9 +1,73 @@
 import { useMutation } from '@apollo/react-hooks';
 import { DELETE_DATETIME } from './dates';
+import { GET_DATETIMES } from '../queries/dates';
+import { GET_TICKETS } from '../queries/tickets';
 
-const useDeleteDatetimeMutation = () => {
+const useDeleteDatetimeMutation = ({ eventId, id }) => {
 	const [deleteDatetime, { loading, error }] = useMutation(DELETE_DATETIME);
-	return { deleteDatetime, loading, error };
+	const variables = { input: { clientMutationId: 'xyz', id } };
+
+	const update = (
+		proxy,
+		{
+			data: {
+				deleteDatetime: { datetime }
+			}
+		}
+	) => {
+		const options = {
+			query: GET_DATETIMES,
+			variables: {
+				where: {
+					eventId
+				}
+			}
+		};
+
+		const { datetimes = {} } = proxy.readQuery(options);
+
+		//TODO - improve readability.
+		const nodes = datetimes.nodes.filter(({ id }) => id !== datetime.id);
+		const data = {
+			datetimes: {
+				...datetimes,
+				nodes
+			}
+		};
+
+		proxy.writeQuery({
+			...options,
+			data
+		});
+
+		if (datetime.id) {
+			const ticketsData = proxy.readQuery({
+				query: GET_TICKETS,
+				variables: {
+					where: {
+						datetimeIn: datetimes.nodes.map(({ id }) => id)
+					}
+				}
+			});
+
+			proxy.writeQuery({
+				query: GET_TICKETS,
+				data: ticketsData,
+				variables: {
+					where: {
+						datetimeIn: nodes.map(({ id }) => id)
+					}
+				}
+			});
```

Comment:
The tickets cache should be updated before updating the datetime cache, otherwise it will trigger refetch for tickets once you update the datetime cache because ticket fetching happens inside `useEffect` with `datetimeIn` being a dependency.

Reply:
Thx @manzoorwanijk updated here: 
https://github.com/eventespresso/event-espresso-core/pull/1897/commits/5cca03d0e234bd0b7f78ff29de915d230eed7d07